### DISCUSSION TITLE
Data Collection

### DISCUSSION THREAD

**Discussion by powmeister** [aristocratos/bpytop#420] [Q&A] [answered]
Is bpytop _collecting data_?

Why does it need to connect whilst piggybacking on python?
🞰htop and others _don’t_ do this.

**Comment by aristocratos** [comment, marked as answer]
@powmeister 
> Is bpytop collecting data?

Bpytop is resource monitor, it kinda relies on collecting data to perform the *monitor* part of that description.

Now your second question I've split up in two parts since it seems a bit confused.

> [...] whilst piggybacking on python?

Bpytop is written in python and python is an interpreted language, it won't run without "piggybacking" on python.


> Why does it need to connect [..]

This shouldn't be a surprise if you read the README.md -> `Shows message in menu if new version is available`

Or looked at the code:

https://github.com/aristocratos/bpytop/blob/ac0f6660be0ad87b23ecb7f371a4a737e1bf017f/bpytop.py#L5017-L5044


And last your statement:

> 🞰htop and others don’t do this.

I can assure you many others do, some even have an entire http server included, like [glances](https://github.com/nicolargo/glances)...

**Reply by powmeister** [reply]
Well, I just graduated from htop, was looking for something “more” and I did not expect this. Thank you for the response, I really appreciate it.
